Features/bootstrap/MainContext.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use QATools\BehatExtension\Context\QAToolsContext;
+
+class MainContext extends QAToolsContext
+{
+
+	/**
+	 * @When /^the user enters (.+?) and (.+?)$/
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->page->fillForm($username, $password);
+	}
+
+	/**
+	 * @Then /^the login should fail$/
+	 */
+	public function loginFail()
+	{
+		$this->page = $this->page->loginFailure();
```

Comment:
That's incorrect approach.

Page provides method to interact with elements on a page. It doesn't have methods, that would deliberately fill the form with data, that will lead to failure/success to allow tests to use them.